### DISCUSSION TITLE
Help with subscribing to data

### DISCUSSION THREAD

**Discussion by FifthWit** [pocketbase/pocketbase#5693] [Q&A] [answered]
so I'm making a discord clone and when I fetch initial messages, its dead simple to filter so they only see messages in their channel but I'm not sure how to do it with a subscribe to data, my initial mesasge request is ```        const messages = await locals.pb.collection('messages').getList(1, 100, {
            filter: `channel = "${channelid}"`,
            sort: '+created',
            expand: 'author'
        });```

but my subscribe logic is more complicated and I dont know how to filter to only messages from the channelid var ```

    onMount(async () => {
        console.log('Component mounted, attempting to subscribe to real-time updates');
        unsubscribe = await pb
            .collection('messages')
            .subscribe('*'
                
            , async ({ action, record }) => {
                if (action === 'create') {
                    console.log(record);
                    const author = await pb.collection('users').getOne(record.author, {});
                    console.log("user", author);
                    record.expand = { author };
                    console.log("FULL RECORD", record);
                    messages = [...messages, record]; // Add new message to the end of the array
                }
                if (action === 'delete') {
                    messages = messages.filter((m) => m.id !== record.id);
                }
            });
    });```

**Comment by ganigeorgiev** [comment]
You can specify the same `filter` and `expand` options as part of the `subscribe` call:

```js
pb.collection('messages').subscribe('*', (e) => {
    ...
}, {
    expand: 'author',
    filter: `channel = "${channelid}"`,
})
```

**Comment by FifthWit** [comment, marked as answer]
thanks so mcuh
